pylops/basicoperators/FirstDerivative.py

```diff
@@ -35,22 +37,33 @@ class FirstDerivative(LinearOperator):
     Notes
     -----
     The FirstDerivative operator applies a first derivative to any chosen
-    direction of a multi-dimensional array.
+    direction of a multi-dimensional array using either a second-order
+    centered stencil or first-order forward/backward stencils.
 
     For simplicity, given a one dimensional array, the second-order centered
     first derivative is:
 
     .. math::
         y[i] = (0.5x[i+1] - 0.5x[i-1]) / dx
 
+    while the first-order forward stencil is:
+
+    .. math::
+        y[i] = (x[i+1] - x[i]) / dx
+
+    and the first-order backward stencil is:
+
+    .. math::
+        y[i] = (x[i] - x[i-1]) / dx
+
     """
-    def __init__(self, N, dims=None, dir=0,
-                 sampling=1., edge=False, dtype='float64'):
+    def __init__(self, N, dims=None, dir=0, sampling=1.,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'dir'](https://app.codacy.com/manual/mrava87/pylops/pullRequest?prid=5670796)